build: fix unknown pragma warning with mingw

Biswa96 opened the pull request:
This checks MSVC toolchain with _MSC_VER macro before adding
    pragma warning directive. It is specific to MSVC and shows
    compiler warning with mingw gcc as following:

    main.c:187: warning: ignoring '#pragma warning ' [-Wunknown-pragmas]
      187 | # pragma warning(suppress : 4996)

zeertzjq commented on src/nvim/main.c:
```diff
@@ -184,7 +184,9 @@ void early_init(mparm_T *paramp)
   ovi.dwOSVersionInfoSize = sizeof(ovi);
   // Disable warning about GetVersionExA being deprecated. There doesn't seem to be a convenient
   // replacement that doesn't add a ton of extra code as of writing this.
+#ifdef _MSC_VER
 # pragma warning(suppress : 4996)
+#endif
   GetVersionEx(&ovi);
```
The pragma will not work if you add an `#endif` after it. I guess this may work:
```suggestion
# ifdef _MSC_VER
# pragma warning(suppress : 4996)
  GetVersionEx(&ovi);
# else
  GetVersionEx(&ovi);
# endif
```

Biswa96 committed:
build: fix unknown pragma warning with mingw

This checks MSVC toolchain with _MSC_VER macro before adding
pragma warning directive. It is specific to MSVC and shows
compiler warning with mingw gcc as following:

main.c:187: warning: ignoring '#pragma warning ' [-Wunknown-pragmas]
  187 | # pragma warning(suppress : 4996)
